src/components/export-select-fields/export-select-fields.jsx

```diff
@@ -0,0 +1,76 @@
+import { Tooltip } from 'hadron-react-components';
+import ExportField from 'components/export-field';
+import styles from './export-select-fields.less';
+import { FIELDS } from 'constants/export-step';
+import createStyler from 'utils/styler.js';
+import React, { PureComponent } from 'react';
+import isEqual from 'lodash.isequal';
+import PropTypes from 'prop-types';
+
+
+const style = createStyler(styles, 'export-select-fields');
+
+const fieldInfoSprinkle = 'The fields displayed are from a sample of documents in the collection. To ensure all fields are exported, add missing field names.';
+
+class ExportSelectFields extends PureComponent {
+  static propTypes = {
+    fields: PropTypes.object.isRequired,
+    exportStep: PropTypes.string.isRequired,
+    updateFields: PropTypes.func.isRequired,
+  };
+
+  // shouldComponentUpdate(nextProps) {
+  //   return !isEqual(this.props.fields, nextProps.fields);
+  // }
+
+  handleFieldCheckboxChange = (evt) => {
+    const fields = Object.assign({}, this.props.fields);
```

Comment:
@imlucas solved the re-render issue, and oh my god was is it the most nuanced react thing.

so what i had before here was:
```js
const fields = this.props.fields;
```
which I then modified and updated the redux state with. So _this is considered a mutation_ by redux, so it will just give up and not fire a re-render apparently \o/

using `Object.assign()` gets rid of the mutation and creates a brand new thing i can rewrite.

Reply:
Ah! Sweet! Nice find